### ISSUE TITLE
Create AsyncAPI/OpenAPI schema parser

### ISSUE THREAD

**Issue by fmvilas** [asyncapi/parser#16]
Create AsyncAPI/OpenAPI schema to JSON Schema Draft 07 converter. It should have a single method as follows:

```go
.Parse(schema []byte) (result json.RawMessage, error)
```

`result` must be a JSON Schema Draft 07-compatible of type `json.RawMessage`.

### Implementation suggestion

We could create this library separated from the parser so other projects can benefit from it too.

**Comment by RobertDiebels** [comment]
I'd like to work on this if possible.

**Comment by fmvilas** [comment]
Absolutely! Just want to remark that this is just for schemas, not the whole OpenAPI/AsyncAPI document.

**Comment by RobertDiebels** [comment]
@fmvilas Great :D ! Alright, I'll have a look at the repo you linked this weekend. If I have any questions I'll let you know. Also, could you assign it to me?

**Comment by fmvilas** [comment]
Done!

**Comment by RobertDiebels** [comment]
@fmvilas I've read the documentation in the repository you linked. Am I right in concluding the aim is to parse an AsyncAPI-spec and convert it to a JSON-schema so other software can use the JSON-schema to validate incoming/outgoing messages?

**Comment by fmvilas** [comment]
You're totally right :)

This way we can leverage existing JSON Schema tooling, which is a lot.

**Comment by RobertDiebels** [comment]
@fmvilas I might take a bit longer to get this fixed. I'm on developing on Windows and I'm encountering some issues when compiling the code-base to C. I'm currently looking for a solution that doesn't involve to much hassle.

**Comment by fmvilas** [comment]
I'd forget about the parser and would do it as a separate library/repo. You don't need the rest of the parser nor compiling it to C to make it work.

Make it simpler and go for a simple Go library that takes a byte array and returns a json.RawMessage and error:

```go
func Parse(protobufSchema []byte) (json.RawMessage, error) {
  ...
}
```

See more here: https://github.com/asyncapi/parser/blob/master/TECHNICAL%20PROPOSAL.md#api-of-a-specific-schema-parser-instance-eg-protobuf-schema-parser

**Comment by Kimbohlovette** [comment]
Hello, @fmvilas. Is this issue up for grabs?  I would like to work on it please.

**Comment by fmvilas** [comment]
Hey @Kimbohlovette! I'm not a maintainer here so honestly have no idea. My guess is that this repo is currently not maintained at all. Maintainers @smoya and @magicmatatjahu can confirm though.

That said, we're always looking for people to jump in and take ownership of repos by contributing. So don't hesitate to contribute if this tool is something you'd love to have.
